src/overlays/actors/ovl_En_Fw/z_en_fw.c

```diff
@@ -21,39 +29,438 @@ const ActorInit En_Fw_InitVars = {
+            this->slideTimer--;
+            if (this->slideTimer == 0) {
+                this->turnAround = true;
+                this->runDirection = -this->runDirection;
+                return;
+            }
+        } else {
+            Math_SmoothScaleMaxMinF(&this->actor.speedXZ, 6.0f, 0.1f, 1.0f, 0.0f);
+            phi_v0 = this->skelAnime.animCurrentFrame;
+            if (phi_v0 == 1 || phi_v0 == 4) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLAME_MAN_RUN);
+                EnFw_SpawnDust(this, 8, 0.16f, 0.1f, 1, 0.0f, 20.0f, 0.0f);
+            }
+        }
+    }
+}
+
+void EnFw_TurnToParentInitPos(EnFw* this, GlobalContext* globalCtx) {
+    s16 angleToParentInit;
+
+    angleToParentInit = Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->actor.parent->initPosRot.pos);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, angleToParentInit, 4, 0xFA0, 1);
+    if (ABS(angleToParentInit - this->actor.shape.rot.y) < 0x65) {
+        // angle to parent init pos is ~0.5 degrees
+        this->actor.posRot.rot = this->actor.shape.rot;
+        this->actor.velocity.y = 14.0f;
+        this->actor.initPosRot.pos = this->actor.posRot.pos;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+        func_80034EC0(&this->skelAnime, D_80A1FBA0, 1);
+        this->actionFunc = EnFw_JumpToParentInitPos;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fw/func_80A1E760.s")
+void EnFw_JumpToParentInitPos(EnFw* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1 && this->actor.velocity.y <= 0.0f) {
+        this->actor.parent->params |= 0x8000;
+        Actor_Kill(&this->actor);
+    } else {
+        Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.x, this->actor.parent->initPosRot.pos.x, 0.6f, 8.0f, 0.0f);
+        Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.z, this->actor.parent->initPosRot.pos.z, 0.6f, 8.0f, 0.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fw/func_80A1E838.s")
+void EnFw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFw* this = THIS;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((this->actor.flags & 0x2000) != 0x2000) {
+        // not attached to hookshot.
+        Actor_MoveForward(&this->actor);
+        func_8002E4B4(globalCtx, &this->actor, 10.0f, 20.0f, 0.0f, 5);
+        this->actionFunc(this, globalCtx);
+        if (this->damageTimer == 0 && this->explosionTimer == 0 && this->actionFunc == EnFw_Run) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fw/func_80A1E8F8.s")
+s32 EnFw_OverrideLimbDraw(GlobalContext* globalContext, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                          void* thisx) {
+    return false;
```

Comment:
so far we havent been using bools for the skelanime callbacks. might be relevant in other places/files in this pr